server/src/main/java/org/opensearch/index/compositeindex/datacube/DimensionFactory.java

```diff
@@ -46,6 +47,8 @@ public static Dimension parseAndCreateDimension(
                 return new NumericDimension(name);
             case KEYWORD:
                 return new KeywordDimension(name);
+            case IP:
```

Comment:
Can we create and use a generic (ORDINAL) dimesnion type for fields using SortedSetDV? This will help with supporting field types beyond core?

Reply:
I'll replace `keyword` with `ordinal` for now. I'll leave IP as it is - I forsee that we can extend IP to handle some cases around parsing etc.

will that work ?